lib/Sabre/Album/AlbumRoot.php

```diff
@@ -156,13 +156,17 @@ public function getLastModified(): int {
 	}
 
 	public function copyInto($targetName, $sourcePath, INode $sourceNode): bool {
-		if ($sourceNode instanceof File) {
-			$sourceId = $sourceNode->getId();
-			$ownerUID = $sourceNode->getFileInfo()->getOwner()->getUID();
-			return $this->addFile($sourceId, $ownerUID);
+		if (!$sourceNode instanceof File) {
+			throw new Forbidden("The source is not a file");
 		}
-		$uid = $this->userId;
-		throw new \Exception("Can't add file to album, only files from $uid can be added");
+
+		$sourceId = $sourceNode->getId();
+		$ownerUID = $sourceNode->getFileInfo()->getOwner()->getUID();
+		if ($ownerUID !== ($uid = $this->userId)) {
```

Comment:
Could you do this assignment outside the condition ?

Reply:
Done